skops/hub_utils/tests/test_hf_hub.py

```diff
@@ -293,6 +312,68 @@ def test_push_download(
         client.delete_repo(repo_id=repo_id, token=HF_HUB_TOKEN)
 
 
+@pytest.mark.network
+@flaky(max_runs=3)
+@pytest.mark.parametrize(
+    "model_func, data, task",
+    [
+        (get_classifier, iris, "tabular-classification"),
+        (get_regressor, diabetes, "tabular-regression"),
+    ],
+    ids=["classifier", "regressor"],
+)
+def test_inference(
+    model_func,
+    data,
+    task,
+    repo_path,
+    destination_path,
+):
+    # test inference backend for classifier and regressor models.
+    client = HfApi()
+
+    model = model_func()
+    model_path = repo_path / "model.pickle"
+
+    with open(model_path, "wb") as f:
+        pickle.dump(model, f)
+
+    version = metadata.version("scikit-learn")
+    init(
+        model=model_path,
+        requirements=[f'scikit-learn="{version}"'],
+        dst=destination_path,
+        task=task,
+        data=data.data,
+    )
+
+    # a model card is needed for inference engine to work.
+    model_card = card.Card(
+        model, metadata=card.metadata_from_config(Path(destination_path))
+    )
+    model_card.save(Path(destination_path) / "README.md")
+
+    user = client.whoami(token=HF_HUB_TOKEN)["name"]
+    repo_id = f"{user}/test-{uuid4()}"
+
+    push(
+        repo_id=repo_id,
+        source=destination_path,
+        token=HF_HUB_TOKEN,
+        commit_message="test message",
+        create_remote=True,
+    )
+
+    X_test = data.data.head(5)
+    y_pred = model.predict(X_test)
+    output = get_model_output(repo_id, data=X_test, token=HF_HUB_TOKEN)
+    assert np.allclose(output, y_pred)
```

Comment:
nit: How about moving the `assert` to the end of the test, so that clean up is performed even if the assert fails? Even better would be a fixture or context manager that does the setup and cleanup but moving the assert would already help.

Reply:
Didn't end up using fixtures for this cause I couldn't figure out how to do them easily, and the pytest docs are really sparse when it comes to using `request`.